packages/services/api/src/modules/cdn/providers/cdn.provider.ts

```diff
@@ -44,7 +53,45 @@ export class CdnProvider {
     throw new HiveError(`CDN is not configured, cannot resolve CDN target url.`);
   }
 
-  generateToken(targetId: string): string {
+  async generateCdnAccess(args: { organizationId: string; projectId: string; targetId: string }) {
+    await this.authManager.ensureTargetAccess({
+      organization: args.organizationId,
+      project: args.projectId,
+      target: args.targetId,
+      scope: TargetAccessScope.REGISTRY_READ,
+    });
+
+    const token = this.legacy_generateToken(args.targetId);
+    const tokenHash = await bcryptjs.hash(token, await bcryptjs.genSalt());
+    const url = this.getCdnUrlForTarget(args.targetId);
+
+    const s3Key = `cdn-legacy-keys/${args.targetId}`;
+
+    const s3Url = [this.s3Config.endpoint, this.s3Config.bucket, s3Key].join('/');
+    const response = await this.s3Config.client.fetch(s3Url, {
+      method: 'PUT',
+      body: tokenHash,
+    });
+
+    if (response.status !== 200) {
+      throw new Error(`Unexpected Status for storing key. (status=${response.status})`);
+    }
+
+    await this.storage.createCDNAccessToken({
+      id: crypto.randomUUID(),
```

Comment:
We don't let the DBMS decide the UUID, as we want to use the same one as within our new tokens that are introduced in a later PR. Conflicts are very unlikely to happen.